cw_evm/lib/evm_chain_wallet.dart

```diff
@@ -0,0 +1,584 @@
+    num amountToEVMChainMultiplier = pow(10, exponent);
+
+    // so far this can not be made with Ethereum as Ethereum does not support multiple recipients
+    if (hasMultiDestination) {
+      if (outputs.any((item) => item.sendAll || (item.formattedCryptoAmount ?? 0) <= 0)) {
+        throw EVMChainTransactionCreationException(transactionCurrency);
+      }
+
+      final totalOriginalAmount = EVMChainFormatter.parseEVMChainAmountToDouble(
+          outputs.fold(0, (acc, value) => acc + (value.formattedCryptoAmount ?? 0)));
+      totalAmount = BigInt.from(totalOriginalAmount * amountToEVMChainMultiplier);
+
+      if (_erc20Balance.balance < totalAmount) {
+        throw EVMChainTransactionCreationException(transactionCurrency);
+      }
+    } else {
+      final output = outputs.first;
+      // since the fees are taken from Ethereum
+      // then no need to subtract the fees from the amount if send all
+      final BigInt allAmount;
+      if (transactionCurrency is Erc20Token) {
+        allAmount = _erc20Balance.balance;
+      } else {
+        allAmount = _erc20Balance.balance -
+            BigInt.from(calculateEstimatedFee(_credentials.priority!, null));
+      }
+      final totalOriginalAmount =
+          EVMChainFormatter.parseEVMChainAmountToDouble(output.formattedCryptoAmount ?? 0);
+      totalAmount = output.sendAll
+          ? allAmount
+          : BigInt.from(totalOriginalAmount * amountToEVMChainMultiplier);
+
+      if (_erc20Balance.balance < totalAmount) {
+        throw EVMChainTransactionCreationException(transactionCurrency);
+      }
+    }
+
+    final pendingEVMChainTransaction = await _client.signTransaction(
+      privateKey: _evmChainPrivateKey,
+      toAddress: _credentials.outputs.first.isParsedAddress
+          ? _credentials.outputs.first.extractedAddress!
+          : _credentials.outputs.first.address,
+      amount: totalAmount.toString(),
+      gas: _estimatedGas!,
+      priority: _credentials.priority!,
+      currency: transactionCurrency,
+      exponent: exponent,
+      contractAddress:
+          transactionCurrency is Erc20Token ? transactionCurrency.contractAddress : null,
+    );
+
+    return pendingEVMChainTransaction;
+  }
+
+  Future<void> _updateTransactions() async {
+    try {
+      if (_isTransactionUpdating) {
+        return;
+      }
+
+      final isProviderEnabled = await checkIfScanProviderIsEnabled();
+
+      if (!isProviderEnabled) {
+        return;
+      }
+
+      _isTransactionUpdating = true;
+      final transactions = await fetchTransactions();
+      transactionHistory.addMany(transactions);
+      await transactionHistory.save();
+      _isTransactionUpdating = false;
+    } catch (_) {
+      _isTransactionUpdating = false;
+    }
+  }
+
+  Future<bool> checkIfScanProviderIsEnabled() async {
+    bool isEtherscanEnabled = (await sharedPrefs.future).getBool("use_etherscan") ?? true;
+    return isEtherscanEnabled;
+  }
```

Comment:
should be abstract and each child should override it

Reply:
Adjusted it